### PR TITLE
Add Puppet enablement procedure

### PR DESCRIPTION
**Pull request by Lennonka**
This is a takeover of #799 

Cherry-pick into:

* [x] Foreman 3.0
* [x] Foreman 3.1

### CODE REVIEW

File: guides/common/assembly_configuring-puppet-in-satellite.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+[role="_abstract"]
+Puppet is an optional component in {ProjectX}.
+If you installed {Project} afresh and you want to continue using Puppet, you have to enable it in your {Project}.
+If you upgraded your {Project} and you want to stop using Puppet, you can disable it and purge Puppet data and content, respectively.
+
+include::modules/proc_enabling-puppet.adoc[leveloffset=+1]
+
+ifdef::satellite[]
+[role="_additional-resources"]
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet-in-satellite.adoc:21]:
```suggestion
[role="configuring-puppet-additional-resources"]
```

File: guides/common/assembly_configuring-puppet-in-satellite.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+== Additional resources
+* See also link:{PuppetDocURL}[Puppet Guide] for details about management of Puppet modules and configuration management with Puppet.
+endif::[]
+
+ifdef::parent-context-of-assembly_configuring-puppet-in-satellite[:context: {parent-context-of-assembly_configuring-puppet-in-satellite}]
```
Comment (melcorr) [inline guides/common/assembly_configuring-puppet-in-satellite.adoc:26]: Because we are an upstream project, we try to avoid having satellite in the link. 
This means we can reuse @maximiliankolb's docs without orcarhino getting into the Satellite docs and vice versa. 
So if you wouldn't mind removing satellite and not including it in future. 
We are not fully there yet across all the guides, but at least this way we will not introduce more. 
I hope this makes sense and sorry for any confusion.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
+[id="proc_enabling-puppet_{context}"]
+= Enabling Puppet Integration with {Project}
+
+A fresh installation of {Project} does not have any Puppet integration configured by default.
+You need to enable the integration as is appropriate for your situation.
+{Project} can be configured to manage and deploy Puppet Server on the {ProjectServer} or on a {SmartProxy}.
```
Comment (melcorr) [inline guides/common/modules/proc_enabling-puppet.adoc:6]:
```suggestion
{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
```
I know it seems unnatural but there is a convention for Satellite in particular not to use the article before Satellite and Capsule, so we don't use it.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc [hunk 4]
```diff
@@ -17,6 +17,10 @@ include::common/assembly_using-satellite-ansible-content-collections.adoc[levelo
 
 include::topics/Users_and_Roles.adoc[]
 
+ifdef::katello,satellite[]
+include::common/assembly_configuring-puppet-in-satellite.adoc[leveloffset=+1]
```
Comment (melcorr) [inline guides/doc-Administering_Red_Hat_Satellite/master.adoc:21]: @maximiliankolb can you confirm the scenarios for this? 
Should be all except orcharhino maybe? 
This is config management rather than content management isn't it, so it is valid for foreman-el?

File: guides/common/assembly_configuring-puppet-in-satellite.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+Puppet is an optional component in {ProjectX}.
+If you installed {Project} afresh and you want to continue using Puppet, you have to enable it in your {Project}.
+If you upgraded your {Project} and you want to stop using Puppet, you can disable it and purge Puppet data and content, respectively.
+
+include::modules/proc_enabling-puppet.adoc[leveloffset=+1]
+
+ifdef::satellite[]
+[role="_additional-resources"]
+== Additional resources
+* See also link:{PuppetDocURL}[Puppet Guide] for details about management of Puppet modules and configuration management with Puppet.
```
Comment (melcorr) [inline guides/common/assembly_configuring-puppet-in-satellite.adoc:23]: Hey @Lennonka 

We **try** to conform to IBM Style for references... so it would be like: 

For more information about configuration management with Puppet, see [link]()

**Review by melcorr — CHANGES_REQUESTED** [review]
Hey @Lennonka 
Thank you so much for this. 
I have added a few comments. I hope they make sense. 
I'm always here to answer questions if you have any.

**Comment by maximiliankolb** [comment]
I will review this on Friday; thanks so far @Lennonka 👍

File: guides/common/assembly_configuring-puppet-in-satellite.adoc (hunk 1 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet-in-satellite.adoc:21]: I'm not sure that this suggestion is a good idea because of our internal build system compliance. I'll do my best to check.

Comment (melcorr) [reply guides/common/assembly_configuring-puppet-in-satellite.adoc:21]: Can you explain [role=""] to me? I don't think I've ever used it.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc (hunk 4 above)
Comment (Lennonka) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:21]: Waiting for comment.

File: guides/common/assembly_configuring-puppet-in-satellite.adoc (hunk 1 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet-in-satellite.adoc]: Some headings have standardized anchors (such as this one), which, I guess, are used for some metadata-related purpose. I'm still a bit fuzzy about the purpose, though.
I only know that this role in particular is standardized by the [assembly template](https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc), which is mentioned in [RH Modularization Guidelines](https://redhat-documentation.github.io/modular-docs/#_additional_resources_4).
That's as much as I can tell you about it at this point.

Comment (Lennonka) [reply guides/common/assembly_configuring-puppet-in-satellite.adoc]: We use the `newdoc` tool ([GH:redhat-documentation/newdoc](https://github.com/redhat-documentation/newdoc)) to generate stubs using those standard templates (assembly, concept, procedure, or reference), which suggest this naming convention.

**Comment by ehelms** [comment]
Looks short and sweet to me, I would like @ares or @ekohl to also take a look.

**Comment by Lennonka** [comment]
@ehelms @ares @ekohl I would like to add verification steps for the user to check that the enablement went well. Any recommendations?

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
+----
+
+.Verification
+
+. Start each step with an active verb.
+
+. Include one command or action per step.
+
+. Use an unnumbered bullet (*) if the procedure includes only one step.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:54]:
```suggestion
```

I have no idea what this means; looks like a template.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc [hunk 7]
```diff
@@ -17,6 +17,10 @@ include::common/assembly_using-satellite-ansible-content-collections.adoc[levelo
 
 include::topics/Users_and_Roles.adoc[]
 
+ifdef::katello,satellite[]
```
Comment (maximiliankolb) [inline guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]:
```suggestion
ifdef::katello,satellite,orcharhino[]
```

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+--foreman-proxy-puppetca true \
+--foreman-proxy-content-puppet true \
+--enable-puppet \
+--puppet-server true \
+--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
+--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
+----
+
+. Enable Puppet on {SmartProxy}s:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:34]:
```suggestion
. Enable Puppet on {SmartProxies}:
```

We have a special attribute for that: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L95

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+--foreman-proxy-puppet true \
+--foreman-proxy-puppetca true \
+--foreman-proxy-content-puppet true \
+--enable-puppet \
+--puppet-server true \
+--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
+--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
+----
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:33]:
```suggestion
```

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+You need to enable the integration as is appropriate for your situation.
+{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
+Additionally, Puppet Server can be deployed externally to {Project} and integrated for reporting, facts and external node classification (ENC).
+
+.Prerequisites
+
+* You installed and configured {Project}.
+
+.Procedure
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:17]:
```suggestion
```

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+= Enabling Puppet Integration with {Project}
+
+[role="_abstract"]
+A fresh installation of {Project} does not have any Puppet integration configured by default.
+You need to enable the integration as is appropriate for your situation.
+{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
+Additionally, Puppet Server can be deployed externally to {Project} and integrated for reporting, facts and external node classification (ENC).
+
+.Prerequisites
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:13]:
```suggestion
```

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 12]
```diff
@@ -0,0 +1,54 @@
+:_module-type: PROCEDURE
+
+[id="proc_enabling-puppet_{context}"]
+= Enabling Puppet Integration with {Project}
+
+[role="_abstract"]
+A fresh installation of {Project} does not have any Puppet integration configured by default.
+You need to enable the integration as is appropriate for your situation.
+{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
+Additionally, Puppet Server can be deployed externally to {Project} and integrated for reporting, facts and external node classification (ENC).
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:10]: Please reword `can be deployed` > `you can deploy...`

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 13]
```diff
@@ -0,0 +1,54 @@
+:_module-type: PROCEDURE
+
+[id="proc_enabling-puppet_{context}"]
+= Enabling Puppet Integration with {Project}
+
+[role="_abstract"]
+A fresh installation of {Project} does not have any Puppet integration configured by default.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:7]:
```suggestion
By default, {Project} does not ...
```

Please reword.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 14]
```diff
@@ -0,0 +1,54 @@
+:_module-type: PROCEDURE
+
+[id="proc_enabling-puppet_{context}"]
+= Enabling Puppet Integration with {Project}
+
+[role="_abstract"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:6]: What does this mean/do? I have only seen this in `guides/common/modules/proc_synchronizing-ansible-collections.adoc` beforce.  I don't think we need this/rely on this. Do you need this for downstream?

Same goes for line one.

If you don't know for sure why, then please remove both instances.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 15]
```diff
@@ -0,0 +1,54 @@
+:_module-type: PROCEDURE
+
+[id="proc_enabling-puppet_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:3]:
```suggestion
[id="Enabling_Puppet_Integration_{context}"]
```

We usually set the anchor to the title + `_{context}`. Unfortunately, we cannot use two attributes in anchors.

File: guides/common/attributes-satellite.adoc [hunk 16]
```diff
@@ -9,6 +9,8 @@
 :PlanningDocURL: {BaseURL}planning_for_red_hat_satellite/index#
 :ProvisioningDocURL: {BaseURL}provisioning_guide/index#
 :UpgradingDocURL: {BaseURL}upgrading_and_updating_red_hat_satellite/index#
+// The following docs are in the downstream only
+:PuppetDocURL: {BaseURL}puppet_guide/index#
```
Comment (maximiliankolb) [inline guides/common/attributes-satellite.adoc:13]: From a first glance, I don't see any "new" guide. If so, you'd need the whole skeleton. See https://github.com/theforeman/foreman-documentation/pull/777 where I have a draft PR to create a new guide (new folder, new symlinks, etc.)

So, IMHO remove both lines.

File: guides/common/assembly_configuring-puppet.adoc [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+
+[role="_abstract"]
+Puppet is an optional component in {ProjectX}.
+If you installed {Project} afresh and you want to continue using Puppet, you have to enable it in your {Project}.
+If you upgraded your {Project} and you want to stop using Puppet, you can disable it and purge Puppet data and content, respectively.
+
+include::modules/proc_enabling-puppet.adoc[leveloffset=+1]
+
+ifdef::satellite[]
+[role="_additional-resources"]
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet.adoc:21]:
```suggestion
```

File: guides/common/assembly_configuring-puppet.adoc [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+[id="assembly_configuring-puppet"]
+endif::[]
+ifdef::context[]
+[id="assembly_configuring-puppet_{context}"]
+endif::[]
+= Configuring Puppet
+
+:context: assembly_configuring-puppet
+
+[role="_abstract"]
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet.adoc:13]:
```suggestion
```

File: guides/common/assembly_configuring-puppet.adoc [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+
+ifndef::context[]
+[id="assembly_configuring-puppet"]
+endif::[]
+ifdef::context[]
+[id="assembly_configuring-puppet_{context}"]
+endif::[]
+= Configuring Puppet
+
+:context: assembly_configuring-puppet
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet.adoc:11]: Why do you want to set the context here? I don't yet see the point or even if this works as expected.

Can we go with the following:

```
[id="Configuring_Puppet_{context}"]
= Configuring Puppet
```

File: guides/common/assembly_configuring-puppet.adoc [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+
+[role="_abstract"]
+Puppet is an optional component in {ProjectX}.
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet.adoc:14]: IMHO there are efforts to not use `{ProjectX}` any longer due to Satellite 6 also omitting the `6`.

File: guides/common/assembly_configuring-puppet.adoc [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+If you installed {Project} afresh and you want to continue using Puppet, you have to enable it in your {Project}.
```
Comment (maximiliankolb) [inline guides/common/assembly_configuring-puppet.adoc:15]: Haven't seen `afresh` in the docs yet, so maybe reword? IMHO we always assume the user has a fresh/plain Foreman installation to begin with.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Made a few comments; however, my biggest question is: Do you really want to add the "enable Puppet" section into the Administering foreman guide, or is "using puppet for configuration management" good enough? IMHO we should keep all the Puppet things together in one place.

Eventually, we could separate both Puppet and Salt and create a stand-alone guide for configuring hosts.

File: guides/common/assembly_configuring-puppet.adoc (hunk 21 above)
Comment (melcorr) [reply guides/common/assembly_configuring-puppet.adoc:15]: I'm pretty sure in the installation guide (at least up until recently), we went with `A fresh installation of {Project}`

Judging by the community behaviour, I think we definitely have to warn people they need to start from scratch. People love to try and install in pretty ripe situations :)

**Comment by melcorr** [comment]
I tend to agree with @maximiliankolb regarding the overall placement of this section. 
This is more to do with host administration and we have a lot of content there about Puppet.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc (hunk 7 above)
Comment (melcorr) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: Can anyone explain to me why we're excluding foreman-el here? 
I am unsure if we even need an ifdef. It seems to apply to all scenarios?

File: guides/common/attributes-satellite.adoc (hunk 16 above)
Comment (melcorr) [reply guides/common/attributes-satellite.adoc:13]: It's referring to this one https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/puppet_guide/index

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc (hunk 7 above)
Comment (maximiliankolb) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: I can't.
If it's in "using Puppet for configuration management", then you'd need to apply the include logic there anyways. Does that work for Satellite?

Comment (melcorr) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: Puppet support is remaining for Satellite to some degree, I don't fully understand it. 
@ehelms can you help us understand a bit better. 
First question: we can just drop the ifdef as this applies everywhere I can tell?
Second question: We have quite an old outdated Satellite guide here: https://access.redhat.com/documentation/en-us/red_hat_satellite/6.10/html/puppet_guide/index - while recently @maximiliankolb created these very nice config management docs here: https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#using-puppet-for-configuration-management_managing-hosts
My question: if there is some degree of Puppet supported for Satellite, could we not just expose these sections for Satellite also?

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
+Additionally, Puppet Server can be deployed externally to {Project} and integrated for reporting, facts and external node classification (ENC).
+
+.Prerequisites
+
+* You installed and configured {Project}.
+
+.Procedure
+
+. Enable Puppet on {ProjectServer} before you enable it on {SmartProxy}s:
```
Comment (melcorr) [inline guides/common/modules/proc_enabling-puppet.adoc:18]:
```suggestion
. Enable Puppet on {ProjectServer} before you enable it on {SmartProxies}:
```

File: guides/common/assembly_configuring-puppet.adoc (hunk 17 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet.adoc:21]: This is a template-given standard that we use for metadata in downstream publishing, therefore it will stay.
See https://github.com/theforeman/foreman-documentation/pull/890#discussion_r760410889

File: guides/common/assembly_configuring-puppet.adoc (hunk 18 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet.adoc:13]: This is a template-given standard that we use for metadata in downstream publishing, therefore it will stay.
See https://github.com/theforeman/foreman-documentation/pull/890#discussion_r760410889

File: guides/common/assembly_configuring-puppet.adoc (hunk 21 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet.adoc:15]: FYI, there will be a second procedure for disabling Puppet after upgrade, so it's also a good for contrasting those two scenarios.

File: guides/common/assembly_configuring-puppet.adoc (hunk 19 above)
Comment (Lennonka) [reply guides/common/assembly_configuring-puppet.adoc:11]: I think that this context is set for the modules included inside the assembly. IMHO the context changes depending on how the files are nested by inclusions, i.e. an assembly should have the context of the parent guide, whereas included modules should have the context of the parent assembly.
But I agree, we need to test it. I'll test if it works with make build.
This is again suggested by Red Hat templates for [docs modularization](https://redhat-documentation.github.io/modular-docs/#assembly-guidelines) and we're going to need it for our Satellite publishing.
So, no, I'm sorry, but with clear conscience, we can't go with your suggestion.

Comment (melcorr) [reply guides/common/assembly_configuring-puppet.adoc:11]: The context is being set for the guide in the master.adoc for each guide it is referenced in. 
We've never had problems building before this, so I tend to also question the need for this.

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc (hunk 7 above)
Comment (ehelms) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: Currently, for foreman-el, Puppet is still enabled by default. So enabling Puppet in this way would not apply to those users. This only applies to Katello/Orcarhino/Satellite for the present moment.

Comment (melcorr) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: Thank you !

Comment (Lennonka) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: Houston, we have a problem.
The "Using Puppet..." assembly is included for all products except Satellite (`ifndef::satellite[]`), so if that condition is correct, then it's a no-go for the intention I had.
I still think it would fit best in the Admin Guide, because it's about project system configuration rather than managing hosts. Enabling Puppet can be a prerequisite for the latter, and xref'd for the sake of Orcharhino and Katello.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+# {foreman-installer} --enable-foreman-plugin-puppet \
+--enable-foreman-cli-puppet \
+--foreman-proxy-puppet true \
+--foreman-proxy-puppetca true \
+--foreman-proxy-content-puppet true \
+--enable-puppet \
+--puppet-server true \
+--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
+--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
```
Comment (ekohl) [inline guides/common/modules/proc_enabling-puppet.adoc:31]: This should be hidden behind `ifdef` and only apply to Katello and its derivatives.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 24]
```diff
@@ -0,0 +1,54 @@
...
+
+. Enable Puppet on {ProjectServer} before you enable it on {SmartProxy}s:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# {foreman-installer} --enable-foreman-plugin-puppet \
+--enable-foreman-cli-puppet \
+--foreman-proxy-puppet true \
+--foreman-proxy-puppetca true \
+--foreman-proxy-content-puppet true \
```
Comment (ekohl) [inline guides/common/modules/proc_enabling-puppet.adoc:26]: This should be hidden behind ifdef and only apply to Katello and its derivatives.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
+----
+
+. Enable Puppet on {SmartProxy}s:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# {foreman-installer} --foreman-proxy-puppet true \
+--foreman-proxy-puppetca true \
+--foreman-proxy-content-puppet true \
```
Comment (ekohl) [inline guides/common/modules/proc_enabling-puppet.adoc:40]: This should be hidden behind ifdef and only apply to Katello and its derivatives.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 26]
```diff
@@ -0,0 +1,54 @@
...
+--enable-puppet \
+--puppet-server true \
+--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
+--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
+--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
```
Comment (ekohl) [inline guides/common/modules/proc_enabling-puppet.adoc:45]: This should be hidden behind ifdef and only apply to Katello and its derivatives.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: guides/doc-Administering_Red_Hat_Satellite/master.adoc (hunk 7 above)
Comment (Lennonka) [reply guides/doc-Administering_Red_Hat_Satellite/master.adoc:20]: I see! Thanks, I'm on it.

File: guides/common/assembly_using-puppet-for-configuration-management.adoc [hunk 27]
```diff
@@ -11,9 +11,16 @@ include::modules/con_puppet-architecture.adoc[leveloffset=+2]
 
 include::modules/proc_prerequisites-for-puppet.adoc[leveloffset=+2]
 
+ifdef::katello,satellite,orcharhino[]
+include::modules/proc_enabling-puppet.adoc[leveloffset=+2]
+endif::[]
+
```
Comment (Lennonka) [inline guides/common/assembly_using-puppet-for-configuration-management.adoc:17]: @ekohl This ifdef ensures that the whole procedure shows up only for Katello and its derivatives. Is that what you meant?

**Comment by Lennonka** [comment]
I've reworked this to include Puppet enablement in Managing Hosts. This is a quickfix that we should be able to deliver for the Satellite 7.0 Testathon in mid-February.

FYI We should follow up with extraction of the "Using Puppet..." chapter into a standalone Puppet Guide, but I suggest leaving it for a separate PR. Debate about this will continue in January 2022.

File: guides/common/modules/proc_enabling-puppet.adoc [hunk 28]
```diff
@@ -0,0 +1,39 @@
+:_module-type: PROCEDURE
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_enabling-puppet.adoc:2]:
```suggestion
```

I would delete the first two lines. If, at some point, we/you need metadata in each procedure, we should figure out a way to do this in one go.

File: guides/common/modules/proc_prerequisites-for-puppet.adoc [hunk 29]
```diff
@@ -1,8 +1,12 @@
 [id="Prerequisites_for_Puppet_{context}"]
-= Prerequisites for Puppet
+= Prerequisites for Puppet Agent
 
-There are four requirements to run the Puppet agent on managed hosts:
+There are the following requirements to run the Puppet agent on managed hosts:
 
+ifdef::katello,satellite,orcharhino[]
+. Enable Puppet integration in your {Project}.
+See xref:Enabling_Puppet_Integration_{context}[].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_prerequisites-for-puppet.adoc:8]:
```suggestion
For more information, see xref:Enabling_Puppet_Integration_{context}[].
```

**Review by maximiliankolb — COMMENTED** [review]
Two small comments & I think Ewoud's comments still stand. The rest LGTM.

File: guides/common/assembly_using-puppet-for-configuration-management.adoc (hunk 27 above)
Comment (ekohl) [reply guides/common/assembly_using-puppet-for-configuration-management.adoc:17]: No, I meant that there are katello-only options in the command. I'll expand a comment on https://github.com/theforeman/foreman-documentation/pull/890/files#r763216790 and others with background info.

File: guides/common/modules/proc_enabling-puppet.adoc (hunk 24 above)
Comment (ekohl) [reply guides/common/modules/proc_enabling-puppet.adoc:20]: The background is that content is a Katello feature. That means any installer option that starts with `--foreman-proxy-content-` is just for Katello. So in this case:
```suggestion
 ifdef::katello,satellite,orcharhino[]
--foreman-proxy-content-puppet true \
 endif::[]
```

File: guides/common/modules/proc_enabling-puppet.adoc (hunk 23 above)
Comment (ekohl) [reply guides/common/modules/proc_enabling-puppet.adoc:25]: Here the background is that Katello has a different SSL certificate setup than regular Foreman. Regular Foreman uses Puppet's certificates so it isn't needed to override this:
```suggestion
ifdef::katello,satellite,orcharhino[]
--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
endif::[]
```

File: guides/common/assembly_using-puppet-for-configuration-management.adoc (hunk 27 above)
Comment (ekohl) [reply guides/common/assembly_using-puppet-for-configuration-management.adoc:17]: Actually, now that I'm reading it, perhaps you're right. In vanilla Foreman Puppet is enabled by default so this whole procedure is not needed.

**Review by melcorr — APPROVED** [review]
LGTM

**Comment by melcorr** [comment]
> LGTM

With Maximilian's feedback applied :o)

**Comment by Lennonka** [comment]
Ewoud's comments are resolved, because the whole module is conditioned for Katello and derivatives, so there's no point in conditioning any of its contents.

**Comment by melcorr** [comment]
Cherry picked to 3.1  041d702 but not to 3.0 because of too many conflicts. 
I think we are OK.
